internal/field_tree_printer/field_tree_printer.go

```diff
@@ -3,48 +3,53 @@ package field_tree_printer
 import (
 	"strings"
 
-	"github.com/codecrafters-io/kafka-tester/internal/field_decoder"
 	"github.com/codecrafters-io/kafka-tester/internal/field_path"
+	"github.com/codecrafters-io/kafka-tester/protocol/value"
 	"github.com/codecrafters-io/tester-utils/logger"
 )
 
+type Field interface {
+	GetPath() field_path.FieldPath
+	GetValue() value.KafkaProtocolValue
+}
+
 type FieldTreePrinter struct {
-	DecodedFields []field_decoder.DecodedField
-	Logger        *logger.Logger
+	Fields []Field
+	Logger *logger.Logger
 
 	currentIndentationLevel int
 	lastPrintedFieldPath    field_path.FieldPath
 }
 
-func (p FieldTreePrinter) PrintForErrorLogs(errorPath field_path.FieldPath) {
+func (p FieldTreePrinter) PrintForErrorLogs(errorPath field_path.FieldPath, errorMessage string) {
 	p.currentIndentationLevel = 0
 	p.lastPrintedFieldPath = field_path.NewFieldPath("")
 
-	for _, decodedField := range p.DecodedFields {
-		p.printNodesLeadingTo(decodedField.Path, p.Logger.Infof)
+	for _, field := range p.Fields {
+		p.printNodesLeadingTo(field.GetPath(), p.Logger.Infof)
 
-		if errorPath.Is(decodedField.Path) {
-			p.Logger.Infof("%s❌ %s (%s)", p.buildIndentPrefix(), decodedField.Path.LastSegment(), decodedField.Value.String())
+		if errorPath.Is(field.GetPath()) {
+			p.Logger.Infof("%s❌ %s (%s)", p.buildIndentPrefix(), field.GetPath().LastSegment(), field.GetValue().String())
 			return
 		} else {
-			p.Logger.Infof("%s- %s (%s)", p.buildIndentPrefix(), decodedField.Path.LastSegment(), decodedField.Value.String())
-			p.lastPrintedFieldPath = decodedField.Path
+			p.Logger.Infof("%s- %s (%s)", p.buildIndentPrefix(), field.GetPath().LastSegment(), field.GetValue().String())
+			p.lastPrintedFieldPath = field.GetPath()
 		}
 	}
 
-	// If ErrorPath didn't match any decoded fields, must be a decode error
+	// If ErrorPath didn't match any decoded fields, must be an error
```

Comment:
This whole function says "ForErrorLogs", so it's obvious there's an error. Reword comment to make more sense